libraries/classes/InsertEdit.php

```diff
@@ -716,6 +718,7 @@ private function getHtmlInput(
             . ' data-type="' . $dataType . '"'
             . ' class="' . $theClass . '" ' . $onChangeClause
             . ' tabindex="' . ($tabindex + $tabindexForValue) . '"'
+            . ($inputMode ? ' ' . $inputMode : '')
```

Comment:
That way the HTML is not stored in variable, so it's more clear what's happening here.

```suggestion
            . ($isInteger ? ' inputmode="numeric"' : '')
```